src/db.js

```diff
@@ -0,0 +1,77 @@
+import Dexie from 'dexie'
+import * as localForage from 'localforage'
+
+const db = new Dexie('EM')
+
+const migrateToDexie = async () => {
+  const THOUGHTINDEX_KEY_START = 'thoughtIndex-'
+  const CONTEXTINDEX_KEY_START = 'contextIndex-'
+  const {
+    cursor,
+    lastUpdated,
+    recentlyEdited,
+    schemaVersion,
+  } = await localForage.getItems([
+    'cursor',
+    'lastUpdated',
+    'recentlyEdited',
+    'schemaVersion',
+  ])
+  const helpersValuesToUpdate = {
+    ...(cursor ? { cursor } : {}),
+    ...(lastUpdated ? { lastUpdated } : {}),
+    ...(recentlyEdited ? { recentlyEdited } : {}),
+    ...(schemaVersion ? { schemaVersion } : {}),
+  }
+  const helperPromise = Object.keys(helpersValuesToUpdate).length ? (async () => db.helpers.put({ id: 'EM', ...helpersValuesToUpdate }))() : {}
+
+  const thoughtIndexes = await localForage.startsWith(THOUGHTINDEX_KEY_START)
+  const thoughtIndexPromises = Object.keys(thoughtIndexes).map(key => {
+    const hash = key.substring(THOUGHTINDEX_KEY_START.length)
+    dbOperations.updateThoughtIndex(hash, thoughtIndexes[key])
+  })
+
+  const contextIndexes = await localForage.startsWith(CONTEXTINDEX_KEY_START)
+  const contextIndexPromises = Object.keys(contextIndexes).map(key => {
+    const hash = key.substring(CONTEXTINDEX_KEY_START.length)
+    dbOperations.updateContextIndex(hash, contextIndexes[key])
+  })
+
+  const migratedItems = await Promise.all([helperPromise, ...thoughtIndexPromises, ...contextIndexPromises])
+  return migratedItems.length
+}
+
+const initHelpers = async () => {
+  const staticHelpersExist = await db.helpers.get({ id: 'EM' })
+  if (!staticHelpersExist) {
+    await db.helpers.add({ id: 'EM' })
+  }
```

Comment:
There can only be one row in the 'helpers' table. So we need to check if it's present before creating it.

Reply:
Do you have ideas for another approach? The additional fetches that have to be `await`'ed are not ideal.

Reply:
I think localStorage would be a better way idea for handling static data of this row. We can do it as a part of this PR or a new issue. Let me know.
I couldn't replicate the render flash that you mentioned. Do you think a render flash is being triggered because of these lines?

Reply:
Not specifically. It's the use of any asynchronous local storage method to retrieve `tutorial` and `fontSize` in particular. When I had them implemented in `localStorage` there was no issue, and then the delay was introduced when we switched to `localForage`. I was hoping `dexie` might reduce the overhead, but it is just using `IndexedDB` so I am not surprised that it is the same.

Let's move the helpers to `localStorage` in a separate PR.

Reply:
I think we can block the render until `tutorial` and `fontSize` are loaded. It sure sounds like a bad approach but a few milliseconds or less won't really have an impact as bad as a render flicker.
Otherwise, localStorage is the way to go.

Reply:
Sorry, I'm not sure I understand. Are you agreeing that we should use `localStorage` here? Otherwise we have a 0.3 - 0.5 sec delay.

Reply:
Do we have a 0.3 - 0.5 sec delay just because of the get call for `tutorial` and `fontsize`?

Reply:
Yes. The initial call to `IndexedDB` appears to have some unavoidable overhead. I haven't tested it extensively though.

Reply:
I can test it. If it causes a significant delay, we can move to localStorage for storing the helpers' table info. We can do that in a separate issue